### ISSUE TITLE
problem on SearchBar with clear button

### ISSUE THREAD

**Issue by ChristianMEC** [react-native-community/react-native-elements#234]
Hello guys,

I am having a problem with clear button on the SearchBar.
In the SearchBar the clear button is clearing the text but is not triggering the **onChangeText** event...

**Comment by betoharres** [comment]
I'm having a issue too.
After setting up the SearchBar like this:
```es6
  <SearchBar lightTheme
        autoCorrect={false}
        onChangeText={props.onSearchBus}
        placeholder={'Search'}
        icon={{name: 'search'}}
        clearIcon={{name: 'clear'}}
        autoFocus={true}
        value={props.searchText} />
```
I'm getting an error when clicking the `clearIcon` saying:
```
undefined is not an object(evaluating `this.refs[ref].clear`)
```

**Comment by binoy14** [comment]
@ChristianMEC I will take a look

@betoharres You need to define `inputRef` for clear function to work. Probably should be documented.

**Comment by betoharres** [comment]
@binoy14 thanks for the heads up! 😄  I don't see it on the README btw. If you want I can make a PR.

After I fixed my bug I can confirm @ChristianMEC issue, it happened here too.

**Comment by binoy14** [comment]
We are using clear method on text input. It should just take care of the onChange event unless we are missing something. 

@betoharres feel free to submit a PR

**Comment by binoy14** [comment]
@ChristianMEC @betoharres This is issue with react native itself. I have filled an issue with react native. Closing this since it does not relate to this repo. 

Ref: https://github.com/facebook/react-native/issues/13251

**Comment by chrise86** [comment]
I'm still having problems with the clear button:

```
<SearchBar
  lightTheme
  ref='searchBar'
  inputRef='searchInput'
  onChangeText={this._filterGroups}
  placeholder='Search groups'
  inputStyle={{backgroundColor: '#fff'}}
  clearIcon={this.state.searchText ? {name: 'close'} : null}
/>
```
I have tried adding `ref` and `inputRef` (as you can see above) but still getting:

    undefined is not an object(evaluating `this.refs[ref].clear`)

**Comment by chrise86** [comment]
It might be worth mentioning here that the required ref is `textInputRef`, not `inputRef`.

**Comment by binoy14** [comment]
Oh sorry, thanks for the correction @chrise86

**Comment by ericwooley** [comment]
facebook/react-native#13251

Got a recommendation to change the supplied code. I agree with the person recommendation. Can we reopen/Fix this?

**Comment by Monte9** [comment]
The fix for this issues was merged as part of this PR: #339. 

Issue should be resolved in the next release `v0.11.3`
